### ISSUE TITLE
`\fontspec_if_feature` not working as expected

### ISSUE THREAD

**Issue by zhouyan** [wspr/fontspec#269]
I think this is a long standing issue or it is the intended behavior.

The macro `\fontspec_if_feature:nTF` does not test feature for the current selected font, instead, it test for the upright font of the current selected font family. Consider the following example,

Consider the following example,

```
\documentclass{article}
\usepackage{fontspec}
\setmainfont{EB Garamond}

\begin{document}

\ExplSyntaxOn

\def\FontName{\lua_now:n{tex.sprint(font.getfont(font.current()).fullname)}}

\def\FontFeatureTF#1#2#3{
  \group_begin:
  \fontspec_set_family:Nnn\thisfont:{}{\FontName}
  \fontfamily\thisfont:\selectfont
  \fontspec_if_feature:nTF{#1}{
    \bool_gset_true:N\g_tmpa_bool
  }{
    \bool_gset_false:N\g_tmpa_bool
  }
  \group_end:
  \bool_if:NTF\g_tmpa_bool{#2}{#3}
}

\itshape
\fontspec_if_feature:nTF{swsh}{Swash}{NoSwash}\break
\FontFeatureTF{swsh}{Swash}{NoSwash}

\ExplSyntaxOff

\end{document}
```

The first test, that directly use `fontspec_if_feature:nTF` shows that  EB Garamond, when italic is selected, does not have swashes, while it does. However, the upright font does not.

The second test, use the macro `FontFeatureTF` as a workaround, correctly shows that it does have. The workaround basically select the current font as a font family on its own, and then test the feature.

**Comment by wspr** [comment]
Thanks — I think you’re right this is a long-term oversight.
(I guess not many people have used the feature!)

I’ll try to fix up shortly. I much appreciate the example code!

**Comment by zhouyan** [comment]
I think even if there are many people use this feature, few will notice this issue, since most of the time, upright and italics have the same features. I have used it for years without noticing it all.

**Comment by wspr** [comment]
This has been addressed (I believe) in e6593dd5e190cebb3c41438edb245f5d92b2e45f. Thanks again!
